### ISSUE TITLE
`IconButton` should have a tooltip by default

### ISSUE THREAD

**Issue by khiga8** [primer/react#3137]
Having a visible label for an IconButton will help users with cognitive disabilities and increase usability for everyone, since it's not always obvious from the icon what action is possible.

In PVC, all `IconButton` default to having a tooltip ([IconButton PVC docs](https://primer.style/view-components/components/beta/iconbutton)). I recall this decision was made after input from accessibility design. Could we follow the same pattern here? 

Relates to: https://github.com/github/accessibility-audits/issues/3171

**Comment by iansan5653** [comment]
Related to (possibly duplicate of):

- https://github.com/primer/react/issues/2008

**Comment by broccolinisoup** [comment]
This might need first releasing the Tooltip remediations https://github.com/primer/react/pull/3169
